dev/format_code.sh

```diff
@@ -182,28 +186,35 @@ xmlsort_failed() {
   exit 7
 }
 echo "Formatting resource files with xmlsort..."
-if [ "$ALL" == "--all" ]; then
-  TRACKED=$(git ls-files -- "**.qrc")
-else
-  # Only files which differ from the master branch
-  TRACKED=$(git diff --name-only master -- "**.qrc")
-fi
-UNTRACKED=$(git ls-files --others --exclude-standard -- "**.qrc")
-for file in $TRACKED $UNTRACKED
+for dir in img/ libs/
 do
-  if [ -f "$file" ]; then
-    OLD_CONTENT=$(cat "$file")
+  for file in $(search_files "${dir}" .qrc)
+  do
     NEW_CONTENT=$(xmlsort -r "RCC/qresource/file" -i -s "$file" || xmlsort_failed)
     NEW_CONTENT="${NEW_CONTENT//\'/\"}"
-    if [ "$NEW_CONTENT" != "$OLD_CONTENT" ]
-    then
-      printf "%s\n" "$NEW_CONTENT" > "$file"
-      echo "[M] $file"
-      COUNTER=$((COUNTER+1))
-    else
-      echo "[ ] $file"
+    update_file "$file"
+  done
+done
+
+# Format qml source files with qmlformat
+qml_format_failed() {
+  echo "" >&2
+  echo "ERROR: qmlformat failed!" >&2
+  echo "  Make sure that qmlformat (qtdeclarative5-dev-tools) are installed." >&2
+  echo "  On Linux, you can also run this script in a docker" >&2
+  echo "  container by using the '--docker' argument." >&2
+  exit 7
+}
+echo "Formatting QML sources with qmlformat..."
+for dir in share/
+do
+  for file in $(search_files "${dir}" .qml)
+  do
+    if [ -x "${QMLFORMAT}" ]; then
```

Comment:
If I understand correctly, this will silently ignore qml files if `qmlformat` is not found. I agree the script should not abort in that case, but I'd highly prefer to see some message in the output that qml files could not be sorted. Isn't this already the case thanks to `qml_format_failed`, so the missing `qmlformat` does not need to be handled separately?

Reply:
> According to error message, I guess to continue rework to enable some formatters on commandline (so user wants only to format qml, not cpp for example). This PR is just test if You agree with changes and I should continue and do it not overengineer ...

So far I agree with the changes, but to be honest I wouldn't add more options to it. Long-term developers shall install `qmlformat` anyway and for short-term developers it's no problem if there appear many warnings in the output. So in both cases we don't need an additional flag (when adding this flag, the question arises why not making all the other tools optional as well, ending up in several more flags not really needed).

Reply:
`qml_format_failed` solves it. :facepalm: thanks.